app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -524,45 +562,79 @@ public void verifyLocationPermissions(Activity activity) {
-
-            @Override
-            public void onClick(View v) {
-
-                emotionalState = String.valueOf(emotionalStateSpinner.getSelectedItem());
-                socialSit = String.valueOf(socialSituationSpinner.getSelectedItem());
-
-                try {
-                    submitMood();
-                    if(editflag) {
-                        Toast.makeText(getApplicationContext(), "Successfully updated!", Toast.LENGTH_SHORT).show();
-                    }
-                    else{
-                        Toast.makeText(getApplicationContext(), "Successfully posted!", Toast.LENGTH_SHORT).show();
-                    }
-                    finish();
-                } catch (DescriptionTooLongException e) {
-                    Toast.makeText(getApplicationContext(), "Your description is too long.", Toast.LENGTH_SHORT).show();
+        emojiList = (LinearLayout) findViewById(R.id.emojiList);
+        for (int i = 1; i <= drawables.size(); i++) {
+
+            LinearLayout emojiLayout = new LinearLayout(this);
+            LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+            params.gravity = Gravity.CENTER_VERTICAL;
+            emojiLayout.setPadding(0,0,20,0);
+            emojiLayout.setOrientation(LinearLayout.VERTICAL);
+            emojiLayout.setTag(i); // This is crucial for grabbing the "index" of the clicked emoji later
+
+            emojiButton = new ImageButton(getApplicationContext());
+            emojiButton.setAdjustViewBounds(true);
+            emojiButton.setMaxHeight(150);
+            emojiButton.setMaxWidth(150);
+            emojiButton.setBackgroundResource(R.color.white);
+            emojiButton.setImageResource(getApplicationContext().getResources().getIdentifier("emoji" + i, "drawable", getApplicationContext().getPackageName()));
+            emojiButton.setPadding(0,0,0,0);
+            emojiLayout.addView(emojiButton);
+            emojiButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    View layout = ((LinearLayout)v.getParent()); // Cast to a View from  ViewParent, since ViewParents don't allow us to use getTag()
+                    int selected = (int) layout.getTag();
+                    selectEmotion(selected);
                 }
+            });
+
+            int emotionID = i % NUM_EMOTIONS;
+            emojiTextview = new TextView(getApplicationContext());
+            if(emotionID == 0) {
+                emotionID = NUM_EMOTIONS;
             }
+            emojiTextview.setText(FeelTripApplication.getEmotionalState(emotionID));
+
+            emojiTextview.setTextSize(15);
+            emojiTextview.setTextColor(Color.GRAY);
+            emojiTextview.setGravity(Gravity.CENTER);
+            emojiTextview.setTypeface(emojiTextview.getTypeface(), Typeface.BOLD);
```

Comment:
Actually, now that I think about it, mutexes would make a lot of sense to have wrapping the processing of mood post indexing. Well damn. Guess I'll add that to our TODO list.